app/models/product.rb

```diff
@@ -1,2 +1,14 @@
 class Product < ActiveRecord::Base
+  has_many :products_wish
+  has_many :wishes, through: :products_wish
+
+  # def self.by_wishes
```

Comment:
Refactor this in the new branch. For now we can give the GUI something to render and play with.